cmake/EkatCreateUnitTestWithAsserts.cmake

```diff
@@ -0,0 +1,77 @@
+include(EkatCreateUnitTest)
+
+# This function creates unit tests to verify that calls to assert()
+# perform as expected. This function is not aware of whether asserts
+# are compiled away or not (i.e., whether the macro NDEBUG is present
```

Comment:
CMake should know if the build is debug or not. We can use `EKAT_IS_DEBUG_BUILD` or `CMAKE_BUILD_TYPE`. We could maybe just have this function do nothing if it's not a debug build, that way the users of this function won't have to remember to guard the call.